src/lib.rs

```diff
@@ -38,8 +38,10 @@ pub use crate::zk_evm::ethereum_types;
 
 use self::utils::*;
 
+pub mod asm_templates;
```

Comment:
this and testing_tracer shoudl be used only for testing. (this way they don't get included/compiled when we build this as a library).
Also, no need for them to be public (as they are not used outside of the crate)

So include them under #[cfg(test)]


```rust
#[cfg(test)]
mod asm_templates

#[cfg(test)]
mod testing_tracer
```

Reply:
Made them private. 
But `#[cfg(test)]` will require to change structure (and `#[cfg(test)]` everywhere in tests). Maybe it should be done in another PR?

Reply:
ok.